### PR TITLE
fix(protocol-designer): fix long step name, step details, and step summary layout issue

### PR DESCRIPTION
**Pull request by syao1226**
fixes RQA-3355

# Overview

Fixing layout issues in `StepFormToolbox`, `StepContainer`, and `StepSummary` when step names, step details, and step summaries are too long, causing the step icon to shrink, the step name and step summaries content to go off-screen, and the step details text to exceed the window size.

Refer to the ticket for details on the current layout issues and the expected design.

### CODE REVIEW

File: protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx [hunk 1]
```diff
@@ -209,8 +210,8 @@ export function StepFormToolbox(props: StepFormToolboxProps): JSX.Element {
         height="calc(100vh - 64px)"
         title={
           <Flex gridGap={SPACING.spacing8} alignItems={ALIGN_CENTER}>
-            <Icon size="1rem" name={icon} />
-            <StyledText desktopStyle="bodyLargeSemiBold">
+            <Icon size="1rem" name={icon} style={{ flexShrink: 0 }} />
```
Comment (koji) [inline protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx]: this works as expected

the following also works
```
minWidth="1rem"
```

cc @jerader @ncdiehl11

Comment (alexjoel42) [reply protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx]: [_RQA-3355.json](https://github.com/user-attachments/files/17482285/_RQA-3355.json)


@syao1226 , 

When I tested on https://sandbox.designer.opentrons.com/protocol_designer-fix-step-name/#/overview the text didn't wrap on the step. 

Screenshot 2024-10-22 at 5 31 58 PM

Comment (syao1226) [reply protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx]: ohh didn't take that into consideration, i'll work on fixing that as well! thanks for the catch, alex.

Comment (syao1226) [reply protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx]: i'll change it to use `minWidth`, it looks cleaner to me. thanks, koji

**Comment by koji** [comment]
@syao1226 I just merged my PR that includes line clamp style in atoms into edge.

import `LINE_CLAMP_TEXT_STYLE` from `atoms` and add `css={LINE_CLAMP_TEXT_STYLE(max_lines)}` to a tag.

File: protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx [hunk 2]
```diff
@@ -208,8 +207,14 @@ export function StepFormToolbox(props: StepFormToolboxProps): JSX.Element {
...
         title={
           <Flex gridGap={SPACING.spacing8} alignItems={ALIGN_CENTER}>
-            <Icon size="1rem" name={icon} />
-            <StyledText desktopStyle="bodyLargeSemiBold">
+            <Icon size="1rem" name={icon} minWidth="1rem" />
+            <StyledText
+              desktopStyle="bodyLargeSemiBold"
+              css={`
+                ${LINE_CLAMP_TEXT_STYLE(2)}
+                word-break: break-all
```
Comment (koji) [inline protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx:215]: `word-wrap: break-word;` didn't work?

Comment (syao1226) [reply protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx:215]: i added `break-all` to handle case of very long words or words connected by symbols like underscore.
Screenshot 2024-10-23 at 1 53 46 PM
Screenshot 2024-10-23 at 1 54 49 PM

Comment (koji) [reply protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepFormToolbox.tsx:215]: thanks!

**Review by koji — APPROVED** [review]
lgtm

**Review by koji — APPROVED** [review]
lgtm

**Pull request merged**